### DISCUSSION TITLE
EpollServerSocketChannel resolve IPV6 address with zone identifier

### DISCUSSION THREAD

**Discussion by srinireddy2020** [netty/netty#14184] [Q&A] [answered]
Dear Folks,
We are using netty (4.1.101) in Apache HBase. Recently while exploring HBASE-27304 jira (where IP will be used in server communication instead of hostname) in IPV6 environment the EpollServerSocketChannel/EpollSocketChannel resolve the remote address with zone ID which is not expected for HBase operations and further leading to connection issues between client and server.
Like eg: 8:10:153:0:0:0:0:15%0 address getting for EpollServerSocketChannel

Is there  any way to avoid zone ID while resolving the IPV6 address?

Note: it works fine when changed to NioServerSocketChannel.

**Comment by normanmaurer** [comment]
So just to clarify you would like to only have it resolve without the scope ID ? So like `8:10:153:0:0:0:0:15`

**Reply by srinireddy2020** [reply]
Yes, Looking for a way EPOLL need to resolve without scope ID

**Reply by normanmaurer** [reply]
when you say resolving how exactly do you construct the channel ? Like what you specify when you bind etc ?

**Reply by normanmaurer** [reply]
I suspect what you mean is that you use `bind("somehostname")` and then `localAddress()` returns the address with scopeId ?

**Reply by srinireddy2020** [reply]
Yes, you suspected correctly.
eg: bootstrap.bind([8:10:153:0:0:0:0:15]:21010).channel().getLocalAddress() returns me "8:10:153:0:0:0:0:15%0"

**Reply by normanmaurer** [reply]
I think this should be fixed by https://github.com/netty/netty/pull/14188

**Reply by srinireddy2020** [reply]
Can we have an API which returns IPV6 address without scope ID?

**Reply by normanmaurer** [reply]
Not sure what you mean with that... `Inet6Address.getScopeId()` will always return something... In case of it not existing it will return 0. https://docs.oracle.com/javase/8/docs/api/java/net/Inet6Address.html#getScopeId--

**Reply by srinireddy2020** [reply]
I meant, to add an API in which just return IPV6 (8:10:153:0:0:0:0:15) without scope ID

**Reply by normanmaurer** [reply, marked as answer]
Like I said this is exactly what it is ... the InetSocketAddress that will be returned has an Inet6Address which will not include the scopeId in the hostname / hostaddress.

**Reply by pankaj72981** [reply]
Thank you so much @normanmaurer for the quick support. What is the release plan for netty-4.1.112?
